### PR TITLE
Dockerfiles: copy RHEL-9 bits over top of unused RHEL8 bits

### PR DESCRIPTION
**Pull request by dcbw**
Step 2 to get back to using regular names (eg without "-rhel-9" suffix). This is posssible without breaking CI because
https://github.com/openshift/release/pull/36772 merged which stops using these Dockerfiles.

Next steps:
- flip the dockerfile_paths in openshift/release back to the non-.rhel9 names since this PR makes the normal names RHEL9 now.
- finally 'git rm' the old .rhel9 Dockefiles in ovnkube.

### CODE REVIEW

**Comment by dcbw** [comment]
/test e2e-aws-ovn-upgrade

AWS failed to provision a node

```
openshift-machine-api   ci-op-8f1ybbq0-e1d2a-nfgvv-master-0                  Running        m6a.xlarge   us-west-1   us-west-1b   27m
openshift-machine-api   ci-op-8f1ybbq0-e1d2a-nfgvv-master-1                  Running        m6a.xlarge   us-west-1   us-west-1a   27m
openshift-machine-api   ci-op-8f1ybbq0-e1d2a-nfgvv-master-2                  Running        m6a.xlarge   us-west-1   us-west-1b   27m
openshift-machine-api   ci-op-8f1ybbq0-e1d2a-nfgvv-worker-us-west-1a-lqbqj   Running        m6a.xlarge   us-west-1   us-west-1a   23m
openshift-machine-api   ci-op-8f1ybbq0-e1d2a-nfgvv-worker-us-west-1b-9mb2d   Provisioning                                         23m
openshift-machine-api   ci-op-8f1ybbq0-e1d2a-nfgvv-worker-us-west-1b-jkprn   Running        m6a.xlarge   us-west-1   us-west-1b   23m
```

**Comment by dcbw** [comment]
/test e2e-metal-ipi-ovn-ipv6


```
fubarhouse.golang failed in directory /root/.ansible/tmp/ansible-
local-79938ihzlv0z5/tmp2k6raml0 (rc=128) - Cloning into 'fubarhouse.golang'...
error: RPC failed; HTTP 500 curl 22 The requested URL returned error: 500
fatal: expected flush after ref listing
```

**Comment by dcbw** [comment]
/test e2e-azure-ovn

Azure failure appears to be:

Failure to delete mirror pod for the controller-manager v4:

```
Mar 07 18:05:11.664195 ci-op-8f1ybbq0-e6384-dvmkb-master-1 kubenswrapper[2314]: I0307 18:05:11.664193    2314 mirror_client.go:131] "Deleting a mirror pod" pod="openshift-kube-controller-manager/kube-controller-manager-ci-op-8f1ybbq0-e6384-dvmkb-master-1" podUID=850e3705-8e6a-4f60-a2c3-6989c7973164
Mar 07 18:05:11.953821 ci-op-8f1ybbq0-e6384-dvmkb-master-1 kubenswrapper[2314]: I0307 18:05:11.953677    2314 kubelet_getters.go:182] "Pod status updated" pod="openshift-kube-controller-manager/kube-controller-manager-ci-op-8f1ybbq0-e6384-dvmkb-master-1" status=Running
Mar 07 18:05:45.667308 ci-op-8f1ybbq0-e6384-dvmkb-master-1 kubenswrapper[2314]: E0307 18:05:45.667264    2314 mirror_client.go:139] "Failed deleting a mirror pod" err="Timeout: request did not complete within requested timeout - context deadline exceeded" pod="openshift-kube-controller-manager/kube-controller-manager-ci-op-8f1ybbq0-e6384-dvmkb-master-1"
Mar 07 18:06:00.756309 ci-op-8f1ybbq0-e6384-dvmkb-master-1 kubenswrapper[2314]: I0307 18:06:00.756271    2314 status_manager.go:693] "Failed to get status for pod" podUID=9889159898cefd25199f7ecd7e0f87bc pod="openshift-kube-controller-manager/kube-controller-manager-ci-op-8f1ybbq0-e6384-dvmkb-master-1" err="the server was unable to return a response in the time allotted, but may still be processing the request (get pods kube-controller-manager-ci-op-8f1ybbq0-e6384-dvmkb-master-1)"
```

**Comment by dcbw** [comment]
/test okd-e2e-gcp-ovn

Same issue with kube-controller-manager

**Comment by dcbw** [comment]
/test e2e-vsphere-windows

```
level=fatal msg=failed to fetch Terraform Variables: failed to fetch dependency of "Terraform Variables": failed to generate asset "Platform Provisioning Check": POST "/sdk": 503 Service Unavailable
```

**Comment by dcbw** [comment]
/test e2e-aws-ovn-local-gateway

```
level=error msg=Error: creating EC2 Instance: InvalidNetworkInterface.InUse: Interface: [eni-031d53436d56bc61a] in use.
level=error msg=	status code: 400, request id: f6afc4f6-9a56-44b9-b19d-ee0e85fe7535
```

**Comment by dcbw** [comment]
/override  ci/prow/4.13-upgrade-from-stable-4.12-e2e-aws-ovn-upgrade 
/override ci/prow/4.13-upgrade-from-stable-4.12-images
/override ci/prow/4.13-upgrade-from-stable-4.12-local-gateway-e2e-aws-ovn-upgrade
/override ci/prow/4.13-upgrade-from-stable-4.12-local-gateway-images

These were left-over from when this PR was against 4.13; for 4.14 they should be updated to 4.13 -> 4.14 upgrade jobs

**Comment by dcbw** [comment]
/override ci/prow/e2e-aws-ovn-upgrade

Kube API budget burn

File: Dockerfile.base [hunk 1]
```diff
@@ -5,23 +5,23 @@
 # The standard name for this image is ovn-kubernetes-base
 
 # build base image shared by both OpenShift and MicroShift
-FROM registry.ci.openshift.org/ocp/4.13:base
+FROM registry.ci.openshift.org/ocp/builder:rhel-9-base-openshift-4.14
```
Comment (tssurya) [inline Dockerfile.base:8]: So qq: on the other PR you linked: https://github.com/openshift/release/pull/36772 if I look at the master.yaml file everything there is still referencing 4.13 and not 4.14. why is that? as in I see `rhel-9-base-openshift-4.13` not `rhel-9-base-openshift-4.14` -> will that happen after this gets in? confused regarding the dependencies here...

File: Dockerfile.base [hunk 2]
```diff
@@ -5,23 +5,23 @@
...
 
 # install selinux-policy first to avoid a race
 RUN dnf install -y --nodocs \
-	selinux-policy && \
+	selinux-policy procps-ng && \
```
Comment (tssurya) [inline Dockerfile.base:12]: nit: we should start adding comments for which packages are needed for what, else not sure how to track the adds/removals for installs dynamically. but this is not related to this PR just a general comment.

File: Dockerfile.base [hunk 3]
```diff
@@ -5,23 +5,23 @@
...
 	dnf clean all
 
-ARG ovsver=2.17.0-62.el8fdp
-ARG ovnver=22.12.0-18.el8fdp
+ARG ovsver=3.1.0-2.el9fdp
+ARG ovnver=23.03.0-4.el9fdp
```
Comment (tssurya) [inline Dockerfile.base:16]: didn't we bump ovn to 23.03 already last week? is the merge bot so smart that it's not detecting conflicts? nice!

**Review by tssurya — APPROVED** [review]
LGTM.
I had a comment regarding the release commit and relation to this PR. once that is answered I can stamp the label onto this.

File: Dockerfile.base (hunk 1 above)
Comment (tssurya) [reply Dockerfile.base:8]: @dcbw ^

File: Dockerfile [hunk 4]
```diff
@@ -35,12 +33,13 @@ ENV PYTHONDONTWRITEBYTECODE yes
 # - openvswitch-ipsec
 # - ovn-vtep
 RUN INSTALL_PKGS=" \
-	openssl python3-pyOpenSSL firewalld-filesystem \
```
[the remark this answers is not in the document]
Comment (dcbw) [reply Dockerfile:38]: We don't need python3-pyopenssl because OVS 3.x moves to Python3's built-in SSL libraries, which in turn use OpenSSL, so it gets pulled in for us automatically and we no longer need to explicitly install it. I don't actually recall why firewalld-filesystem gets removed, but all it creates is some directories in `/usr/lib/firewalld` and presumably we don't need that.

File: Dockerfile.base (hunk 3 above)
Comment (dcbw) [reply Dockerfile.base:16]: Yeah, I sumbitted this PR before the OVN bump landed to the `rhel9` bits, so I should rebase this PR.

File: Dockerfile.base (hunk 1 above)
Comment (dcbw) [reply Dockerfile.base:8]: I think what we'll do is let https://github.com/openshift/ovn-kubernetes/pull/1575 merge and then I'll rebase this PR to match. Sound OK?

Comment (tssurya) [reply Dockerfile.base:8]: yup sgtm.

**Review by tssurya — APPROVED** [review]
/lgtm
assuming we still use the rhel9 dockerfile for now ? after this merges we probably delete that and pivot to this now that this has been cleaned up!

**Comment by dcbw** [comment]
/override ci/prow/4.13-upgrade-from-stable-4.12-local-gateway-images
/override ci/prow/4.13-upgrade-from-stable-4.12-local-gateway-e2e-aws-ovn-upgrade
/override ci/prow/4.13-upgrade-from-stable-4.12-images 
/override ci/prow/4.13-upgrade-from-stable-4.12-e2e-aws-ovn-upgrade

These jobs should  not be running here, as they are 4.12 -> 4.13 upgrades; we are working on moving them to 4.13 -> 4.14

**Comment by dcbw** [comment]
/override ci/prow/e2e-aws-ovn-upgrade
/override ci/prow/e2e-aws-ovn-upgrade-local-gateway

The dockerfiles modified by this PR are not used at this time, so failure is unrelated.
